deltacat/logs.py

```diff
@@ -19,6 +24,45 @@
 DEFAULT_DEBUG_DELTACAT_LOG_BASE_FILE_NAME = "deltacat-python.debug.log"
 
 
+class RayRuntimeContextLoggerAdapter(logging.LoggerAdapter):
+    """
+    Logger Adapter for injecting Ray Runtime Context into logging messages.
+    """
+    def __init__(self, logger: Logger, runtime_context: RuntimeContext):
+        super().__init__(logger, {})
+        self.runtime_context = runtime_context
+
+    def process(self, msg, kwargs):
+        """
+        Injects Ray Runtime Context details into each log message.
+
+        This may include information such as the raylet node ID, task/actor ID, job ID,
+        placement group ID of the worker, and assigned resources to the task/actor.
+
+        Args:
+            msg: The original log message
+            kwargs: Keyword arguments for the log message
+
+        Returns: A log message with Ray Runtime Context details
+
+        """
+        runtime_context_dict = self.runtime_context.get()
+        if self.runtime_context.get_task_id() or self.runtime_context.get_actor_id():
+            runtime_context_dict["pg_id"] = self.runtime_context.get_placement_group_id()
+            runtime_context_dict["assigned_resources"] = self.runtime_context.get_assigned_resources()
```

Comment:
with placement group, is this `get_assigned_resources` still able to return the exact node id? If it's just some bundle information, we may need to parse it with pg api, but not sure how much overhead it will add.

Reply:
In the example output I provided, the `assigned_resources` does record the node host ID, is that what you are referring to? It also records the bundle information (e.g. 'CPU: 1.0'). The `get_assigned_resources` will pull details from the [raylet layer](https://github.com/ray-project/ray/blob/master/python/ray/_raylet.pyx#L2240).

Reply:
Runtime Context API: https://docs.ray.io/en/latest/ray-core/package-ref.html#runtime-context-apis

Reply:
sounds good.